### PR TITLE
network.c: simplify get_ip_string()

### LINKED ISSUE #3: warnings during build
```
network.c: In function 'get_ip_string':
network.c:262:53: warning: cast increases required alignment of target type [-Wcast-align]
                         struct sockaddr_in *sa_in = (struct sockaddr_in *) sa;
                                                     ^
network.c:270:29: warning: cast increases required alignment of target type [-Wcast-align]
                             (struct sockaddr_in6 *) sa;
```

### PR DESCRIPTION
**Pull request by rofl0r**
fixes warnings "cast increases required alignment of target type"
fixes unhandled error return from inet_ntop()
decreases code size since function call is reworked from 2 call sites to 1

closes #3

### CODE REVIEW

File: src/network.c [hunk 1]
```diff
@@ -249,35 +249,20 @@ ssize_t readline (int fd, char **whole_buffer)
...
-                }
-        default:
-                /* no valid family */
-                return NULL;
-        }
-
-        return buf;
+	af = sa->sa_family;
+	if(af != AF_INET && af != AF_INET6) return NULL;
+	addr = (af == AF_INET ?
```
Comment (obnoxxx) [inline src/network.c]: I am not a big fan of these ( ... ? ... :  ... ) constructs. It makes the code much harder to read and debug imho.
I like clear and explicit albeit more verbose switch or if / else if / else statements much better.

So if you want to force the use of only one inet_ntop call, i'd prefere an if/else for that.

File: src/network.c [hunk 2]
```diff
@@ -249,35 +249,20 @@ ssize_t readline (int fd, char **whole_buffer)
...
+	      (void*)&(((struct sockaddr_in *) (void*)sa)->sin_addr) :
```
Comment (obnoxxx) [inline src/network.c]: Ok. these casts and alignment warnings drive me crazy.
Could you paste the exact compiler warning you are getting?

File: src/network.c [hunk 3]
```diff
@@ -249,35 +249,20 @@ ssize_t readline (int fd, char **whole_buffer)
...
+	      (void*)&(((struct sockaddr_in6 *) (void*)sa)->sin6_addr) );
+	return inet_ntop(af, addr, buf, buflen);
```
Comment (obnoxxx) [inline src/network.c]: Alright the catch with the ignored return of inet_ntop is good!
(Could also be solved by setting a return variable in the two separate calls in the switch statement.)

**Comment by obnoxxx** [comment]
Ok. Made a couple of comments above.
In general, I think the patch wants to solve too many things at a time.
(A commit message that lists the achievements of the patch in a bullet list is an almost certain indicator for that.)
I usually prefer more atomic patches, and have more of them.

So I am not quite ok with the patch in the current form, but I see where you are going and that looks good!

File: src/network.c (hunk 2 above)
Comment (obnoxxx) [inline src/network.c]: Forget the question. Issue #3 has the paste. :-)

**Comment by rofl0r** [comment]
i agree that the code does several things at once, but the original function was so utterly verbose that i think a rewrite was justified.

**Comment by rofl0r** [comment]
would you accept the PR when i replace the ternary op with an if/else ?

**Comment by obnoxxx** [comment]
I don't object a rewrite. But I request it to land in several patches, each changing one aspect at a time:
- ignored return value of inet_ntop()
- fix cast - alignment warning
- etc

All those can go into this single PR.

Also sometimes, more but more explicit code is better (more clear) than very condensed code. imho

**Comment by obnoxxx** [comment]
As an example, here is what I imagine as patches for fixing the return type and honoring the result of inet_ntop. Simple, atomic patches that fix one piece as a time, in logical easy-to-understand and easy-to-verify units: PR#86

**Comment by rofl0r** [comment]
closing since you have the important bits now in #86

**Comment by obnoxxx** [comment]
Alright thanks! Whenever we hit the alignment problem again, we can resurrect this topic.

**Pull request closed without merging**